test/common/event/scaled_range_timer_manager_test.cc

```diff
@@ -0,0 +1,486 @@
+
+  EXPECT_CALL(callback, Call());
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+  EXPECT_FALSE(timer->enabled());
+}
+
+TEST_F(ScaledRangeTimerManagerTest, EnableAndDisableTimer) {
+  ScaledRangeTimerManager manager(dispatcher_);
+
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+
+  timer->enableTimer(std::chrono::seconds(5), std::chrono::seconds(30));
+  EXPECT_TRUE(timer->enabled());
+
+  timer->disableTimer();
+  EXPECT_FALSE(timer->enabled());
+
+  // Provide some additional guarantee of safety by running the dispatcher for a little bit. This
+  // should be a no-op, and if not (because a timer was fired), that's a problem that will be caught
+  // by the strict mock callback.
+  simTime().advanceTimeAndRun(std::chrono::seconds(10), dispatcher_, Dispatcher::RunType::Block);
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableWhileDisabled) {
+  ScaledRangeTimerManager manager(dispatcher_);
+
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+
+  EXPECT_FALSE(timer->enabled());
+  timer->disableTimer();
+
+  EXPECT_FALSE(timer->enabled());
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableWhileWaitingForMax) {
+  ScaledRangeTimerManager manager(dispatcher_);
+
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+  timer->enableTimer(std::chrono::seconds(10), std::chrono::seconds(100));
+  EXPECT_TRUE(timer->enabled());
+
+  timer->disableTimer();
+  EXPECT_FALSE(timer->enabled());
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableWhileActive) {
+  ScaledRangeTimerManager manager(dispatcher_);
+
+  MockFunction<TimerCb> callback;
+  auto timer = manager.createTimer(callback.AsStdFunction());
+
+  timer->enableTimer(std::chrono::seconds(5), std::chrono::seconds(100));
+
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+
+  EXPECT_TRUE(timer->enabled());
+
+  timer->disableTimer();
+  EXPECT_FALSE(timer->enabled());
+
+  // Run the dispatcher to make sure nothing happens when it's not supposed to.
+  simTime().advanceTimeAndRun(std::chrono::seconds(100), dispatcher_, Dispatcher::RunType::Block);
+}
+
+TEST_F(ScaledRangeTimerManagerTest, DisableFrontActiveTimer) {
+  ScaledRangeTimerManager manager(dispatcher_);
+
+  MockFunction<TimerCb> callback1, callback2;
+  auto timer1 = manager.createTimer(callback1.AsStdFunction());
+  auto timer2 = manager.createTimer(callback2.AsStdFunction());
+
+  // These timers have the same max-min.
+  timer1->enableTimer(std::chrono::seconds(5), std::chrono::seconds(30));
+  timer2->enableTimer(std::chrono::seconds(10), std::chrono::seconds(35));
+
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+  simTime().advanceTimeAndRun(std::chrono::seconds(5), dispatcher_, Dispatcher::RunType::Block);
+
+  timer1->disableTimer();
```

Comment:
It seems that you want to cover behavior in cases where the timer scaling max which is responsible for the queue timer is disabled, and check that the next timer in the queue takes over.  There are no assertions on changes to the queue timer.

Possible suggestion: Advance time to T=30 and verify that nothing triggers, move to T=35 and verify that a timer fires.